File: beginner_tutorials/scripts/simple_action_client.py
#!/usr/bin/env python

import roslib; roslib.load_manifest('beginner_tutorials')
import rospy
import actionlib
from beginner_tutorials.msg import *

if __name__ == '__main__':
	rospy.init_node('do_dishes_client')
	client = actionlib.SimpleActionClient('do_dishes', DoDishesAction)
	client.wait_for_server()

	goal = DoDishesGoal()
	goal.dishwasher_id = 1
	print "Requesting dishwasher %d"%(goal.dishwasher_id)

	client.send_goal(goal)
	client.wait_for_result(rospy.Duration.from_sec(5.0))

	result = client.get_result()
	print "Resulting dishwasher %d"%(result.total_dishes_cleaned)

File: beginner_tutorials/src/simple_action_client.cpp
#include <beginner_tutorials/DoDishesAction.h>
#include <actionlib/client/simple_action_client.h>

typedef actionlib::SimpleActionClient<beginner_tutorials::DoDishesAction> Client;

int main(int argc, char **argv)
{
    ros::init(argc, argv, "do_dishes_client");

    Client client("do_dishes", true);
    client.waitForServer();
    beginner_tutorials::DoDishesGoal goal;

    client.sendGoal(goal);
    client.waitForResult(ros::Duration(5.0));
    if (client.getState() == actionlib::SimpleClientGoalState::SUCCEEDED)
        printf("Yay! The dishes are now clean");

    printf("Current State: %s\n", client.getState().toString().c_str());

    return 0;
}
